homeassistant/components/ps4/media_player.py

```diff
@@ -0,0 +1,390 @@
+    games_file = hass.config.path(GAMES_FILE)
+    creds = config.data[CONF_TOKEN]
+    for device in config.data['devices']:
+        host = device.get(CONF_HOST)
+        region = device.get(CONF_REGION)
+        name = device.get(CONF_NAME)
+        ps4 = pyps4.Ps4(host, creds)
+        add_devices([
+            PS4Device(name, config, host, region, ps4, games_file)], True)
+
+
+class PS4Data():
+    """Init Data Class."""
+
+    def __init__(self):
+        """Init Class."""
+        self.devices = []
+
+
+class PS4Device(MediaPlayerDevice):
+    """Representation of a PS4."""
+
+    def __init__(self, name, config, host, region, ps4, games_file):
+        """Initialize the ps4 device."""
+        self._config = config
+        self._ps4 = ps4
+        self._host = host
+        self._name = name
+        self._region = region
+        self._state = STATE_UNKNOWN
+        self._games_filename = games_file
+        self._media_content_id = None
+        self._media_title = None
+        self._media_image = None
+        self._source = None
+        self._games = self.load_games()
+        self._source_list = list(sorted(self._games.values()))
+        self._retry = 0
+        self._info = None
+        self._unique_id = None
+
+    async def async_added_to_hass(self):
+        """Subscribe PS4 events."""
+        self.hass.data[PLATFORM].devices.append(self)
+
+    @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
+    def update(self):
+        """Retrieve the latest data."""
+        try:
+            status = self._ps4.get_status()
+            if self._info is None:
+                self.get_device_info(self._config, status)
+        except socket.timeout:
+            status = None
+        if status is not None:
+            self._retry = 0
+            if status.get('status') == 'Ok':
+                titleid = status.get('running-app-titleid')
+                name = status.get('running-app-name')
+                if titleid and name is not None:
+                    self._state = STATE_PLAYING
+                    if self._media_content_id != titleid:
+                        self._media_content_id = titleid
+                        self.get_title_data(titleid, name)
+                else:
+                    self.idle()
+            else:
+                self.state_off()
+        elif self._retry > 5:
+            self.state_unknown()
+        else:
+            self._retry += 1
+            self.update()
+
+    def idle(self):
+        """Set states for state idle."""
+        self.no_title()
+        self._state = STATE_IDLE
+
+    def state_off(self):
+        """Set states for state off."""
+        self.no_title()
+        self._state = STATE_OFF
+        self.update()
+
+    def state_unknown(self):
+        """Set states for state unknown."""
+        self.no_title()
+        self._state = STATE_UNKNOWN
+        _LOGGER.debug("PS4 could not be reached")
+        self._retry = 0
+        self.update()
```

Comment:
Why do we continue calling update after the retries have failed? Won't this interfere with the normal update interval?

Reply:
I figured if the PS4 dropped connection it would  be a faster update but I guess it would just throw a bunch of errors and overload the thread. I'll remove.